### PR TITLE
[relay][frontend] Return module from frontend parsers

### PR DESCRIPTION
**Pull request by zhiics**
#3346 

This PR returns `Module` instead of `Expr` from the frontend converters because:
1) As we move to pass manager the optimizations should usually be `Module` to `Module`
2) If the converters need to work on modules, i.e. containing more than one functions, returning `Expr` could be error-prone. Or we should really return a function with closures. Returning module should be more convenient.

Necessary changes are made in various executors to take both module and expr.

Covered converters:

- [x] TensorFlow

- [x] TFLite

- [x] Keras

- [x] Caffe2

- [x] MXNet

- [x] Darknet

- [x] CoreML

- [x] ONNX

### CODE REVIEW

**Comment by zhiics** [comment]
cc @icemelon9 @jroesch @wweic @tqchen @yongwww @yzhliu, plz review.

**Comment by tqchen** [comment]
please rebase against the current master

File: python/tvm/relay/backend/vm.py [hunk 1]
```diff
@@ -131,9 +131,11 @@ def __init__(self, mod, ctx, target):
         self.ctx = ctx
         self.target = target
 
-    def _make_executor(self, expr):
-        assert isinstance(expr, Expr)
-        self.mod[self.mod.entry_func] = expr
+    def _make_executor(self, expr=None):
+        expr = expr if expr else self.mod
```
Comment (jroesch) [inline python/tvm/relay/backend/vm.py]: I think we should try to factor all this normalization code into one spot.

**Comment by tqchen** [comment]
@zhiics please rebase against the latest master

File: python/tvm/relay/backend/vm.py (hunk 1 above)
Comment (tqchen) [reply python/tvm/relay/backend/vm.py]: IMO we should only take module and remove expr related code

Comment (zhiics) [reply python/tvm/relay/backend/vm.py]: I also think we probably only should just take a module, but it looks we need to refactor the interpreter and the graph_runtime somehow.

**Comment by tqchen** [comment]
Thanks, @jroesch @zhiics @icemelon9 , this PR is now merged
